crates/iceberg/src/scan/mod.rs

```diff
@@ -348,19 +426,25 @@ impl TableScan {
         // used to stream the results back to the caller
         let (file_scan_task_tx, file_scan_task_rx) = channel(concurrency_limit_manifest_entries);
 
-        let (delete_file_idx, delete_file_tx) = DeleteFileIndex::new();
-
-        let manifest_list = plan_context.get_manifest_list().await?;
+        // For incremental scan, disable delete file processing
```

Comment:
If we processed delete files, your PR would essentially almost be producing a changelog, like in Spark, right? Perhaps not actionable right now, but mentioning this as a possible extension in future.